app/main/views/g_cloud.py

```diff
@@ -424,17 +424,48 @@ def save_search(framework_family):
     filters = filters_for_lot(current_lot_slug, content_manifest, all_lots=framework['lots'])
     clean_request_query_params = clean_request_args(search_query, filters.values(), lots_by_slug)
 
-    form = CreateProjectForm()
-    name = form.name.data
+    projects = get_direct_award_projects(data_api_client, current_user.id, 'open_projects', 'name')
+    projects.sort(key=lambda x: x['name'])
 
-    save_search_selection = request.form.get('save_search_selection', None)
+    default_selection = {"save_search_selection": "new_search"} if not projects else {}
+    form = CreateProjectForm(projects, data=default_selection)
 
-    name_is_invalid = save_search_selection == "new_search" and not name
+    if form.validate_on_submit():
+        if form.save_search_selection.data == "new_search":
+            try:
+                api_project = data_api_client.create_direct_award_project(user_id=current_user.id,
+                                                                          user_email=current_user.email_address,
+                                                                          project_name=form.name.data)
+            except HTTPError as e:
+                abort(e.status_code)
+
+            project = api_project['project']
+        else:
+            project = data_api_client.get_direct_award_project(project_id=form.save_search_selection.data)['project']
+            if not project or not is_direct_award_project_accessible(project, current_user.id):
+                abort(404)
 
-    if request.method == 'GET' or not form.validate_on_submit() or not save_search_selection or name_is_invalid:
-        projects = get_direct_award_projects(data_api_client, current_user.id, 'open_projects', 'name')
-        projects.sort(key=lambda x: x['name'])
+        search_api_url = search_api_client.get_search_url(
+            index=framework['slug'],
+            **build_search_query(search_query, filters.values(), content_manifest, lots_by_slug)
+        )
+        try:
+            data_api_client.create_direct_award_project_search(user_id=current_user.id,
+                                                               user_email=current_user.email_address,
+                                                               project_id=project['id'],
+                                                               search_url=search_api_url)
+
+        except HTTPError as e:
+            abort(e.status_code)
 
+        flash(PROJECT_SAVED_MESSAGE, 'success')
+
+        return redirect(url_for('.view_project',
+                                framework_family=framework_family,
+                                project_id=project['id']
+                                ), code=303)
```

Comment:
I know this was here before, but why have a `303` status? 🤔

Reply:
[git blame](https://github.com/alphagov/digitalmarketplace-buyer-frontend/commit/31fe0ed76093264e0ed4110cc301205440dcc75f) suggests that it is more appropriate for a POST redirect, it's a HTTP/1.1 feature.

I think it's fine either way, maybe something to look into more in future?

Reply:
Fair enough. I was just curious as we have quite a few other POST redirects in that module that use the default (302, I think).